### ISSUE TITLE
selective subscribe and state await

### ISSUE THREAD

**Issue by drcmda** [developit/unistore#19]
Could this be in this lib? For each and every redux-like lib i end up having to write this stuff again, and i'll be doing it for this as well. But these aren't edge-cases. To get a ping when a certain slice of state changes is something one would use all the time, that is if it isn't entirely driven with connected views but also data models.

```js
function subscribe(selector, callback) {
    let state = selector(store.getState())
    return store.subscribe(() => {
        let selected = selector(store.getState()),
            old = state
        if (typeof selected === 'object' && !Array.isArray(selected)) {
            selected = Object.entries(selected).reduce(
                (acc, [key, value]) => (state[key] !== value ? { ...acc, [key]: value } : acc),
                state,
            )
        }
        if (state !== selected) callback((state = selected), old)
    })
}

function awaitState(selector, condition) {
    return new Promise(resolve => {
        const unsub = subscribe(store, selector, props => {
            const result = condition(props)
            if (result) unsub() || resolve(result)
        })
    })
}


// Example #1: wait for something to come up
const worker = await awaitState(state => state.collections.workers, workers => workers.find(key => key === ID))

// Example #2: Responding to state changes
const unsub = subscribe(state => !state.workers[ID].alive, () => console.log("do somethig"))
```

**Comment by developit** [comment]
Neat! I'm open to adding this if it's plausible without much of a size increase.  One issue I can see is that the subscriptions array needs to include the slice.  Also, this only works if state is treated as immutable, which isn't true for the rest of the library.

**Comment by drcmda** [comment]
@developit  Yes, it would be able to pluck a property or a keyed mapStateToProps object. mstp goes through a reducer, the prop doesn't need that. Both cases would be tested with reference equality. But either way, i'd use whatever unistore gives. No big deal if direct props wouldn't work.

The other point about the subscription-array, i don't follow. The subscribe simply drops a scope bound state and checks against it. It doesn't need to expose the slice externally. Doesn't connect work in a similar way?

As for size, given that it probably would be feasible to re-use connects subscriber, it would only add a couple of bytes.

**Comment by developit** [comment]
If there is a way to extra & reuse the logic from connect so the size stays down I'd be up for it!
